sampledata/l10n/skills.py

```diff
@@ -60,4 +64,12 @@ def generate(self, sd, locale=None, subtype=None):
         else:
             skills = self.all_skills()
 
-        return sd.choice(skills)
+        random.shuffle(skills)
+
+        if total:
+            if not isinstance(total, int):
+                raise ParameterError('Not valid total')
+            return skills[0:total]
```

Comment:
Sorry, but I don't get what you mean:

>>> a = [1,2,3,4]
>>> a[0:1000]
[1, 2, 3, 4]



On 24 September 2015 at 10:46, Jesús Espino <notifications@github.com>
wrote:

> In sampledata/l10n/skills.py
> <https://github.com/jespino/sampledata/pull/6#discussion_r40296968>:
>
> > @@ -60,4 +64,12 @@ def generate(self, sd, locale=None, subtype=None):
> >          else:
> >              skills = self.all_skills()
> >
> > -        return sd.choice(skills)
> > +        random.shuffle(skills)
> > +
> > +        if total:
> > +            if not isinstance(total, int):
> > +                raise ParameterError('Not valid total')
> > +            return skills[0:total]
>
> This can raise an out of range exception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jespino/sampledata/pull/6/files#r40296968>.
>



-- 
Yamila  Moreno Suárez
http://dendarii.es
http://moduslaborandi.net